### PR TITLE
fix xyz for 1.9

### PR DESCRIPTION
**Pull request by w-e-w**
ther was a change in 1.9 that cases this extention to break
- cause https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/23c06a51ccc54d76db465f8f66a5dcb0f7ca33e5
- fix refernce https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/15532

### CODE REVIEW

**Comment by w-e-w** [comment]
not sure what is this lint
https://github.com/Bing-su/adetailer/actions/runs/8779110515

**Comment by w-e-w** [comment]
Update
AUTO as pushed a fix to issue in webui 1.9.2 reverting the behavior back to 1.8
see https://github.com/AUTOMATIC1111/stable-diffusion-webui/releases/tag/v1.9.2

this PR now not as important and only applicable to those that are on 1.9.0 and did not updata to 1.9.2
if merged this will fix the for 1.9.0 but if not, it will still work for <=1.8 and >=1.9.2

**Comment by Bing-su** [comment]
Thanks for your attention to this issue, but I'm questioning whether this issue actually existed.

Below is logs from 1.9.0, which I haven't updated yet.

```sh
Python 3.10.14 | packaged by conda-forge | (main, Mar 20 2024, 12:40:08) [MSC v.1938 64 bit (AMD64)]
Version: v1.9.0-112-gd982c75d
Commit hash: d982c75da9a1647770b3226edaa0a603d20e3a8b
Launching Web UI with arguments: --xformers --api
Tag Autocomplete: Could not locate model-keyword extension, Lora trigger word completion will be limited to those added through the extra networks menu.
[-] ADetailer initialized. version: 24.4.2, num models: 13
ControlNet preprocessor location: D:\stable-diffusion-webui\extensions\sd-webui-controlnet\annotator\downloads
2024-04-22 19:49:06,465 - ControlNet - INFO - ControlNet v1.1.443
2024-04-22 19:49:06,607 - ControlNet - INFO - ControlNet v1.1.443
Loading weights [5998292c04] from D:\stable-diffusion-webui\models\Stable-diffusion\Counterfeit-V3.0_fp16-no-ema.safetensors
extensions\adetailer\scripts\!adetailer.py:939 make_axis_on_xyz_grid
    script.script_class.__module__: 'custom_code.py' (str) len=14
extensions\adetailer\scripts\!adetailer.py:939 make_axis_on_xyz_grid
    script.script_class.__module__: 'img2imgalt.py' (str) len=13
extensions\adetailer\scripts\!adetailer.py:939 make_axis_on_xyz_grid
    script.script_class.__module__: 'loopback.py' (str) len=11
extensions\adetailer\scripts\!adetailer.py:939 make_axis_on_xyz_grid
    script.script_class.__module__: 'outpainting_mk_2.py' (str) len=19
extensions\adetailer\scripts\!adetailer.py:939 make_axis_on_xyz_grid
    script.script_class.__module__: 'poor_mans_outpainting.py' (str) len=24
extensions\adetailer\scripts\!adetailer.py:939 make_axis_on_xyz_grid
    script.script_class.__module__: 'prompt_matrix.py' (str) len=16
extensions\adetailer\scripts\!adetailer.py:939 make_axis_on_xyz_grid
    script.script_class.__module__: 'prompts_from_file.py' (str) len=20
extensions\adetailer\scripts\!adetailer.py:939 make_axis_on_xyz_grid
    script.script_class.__module__: 'sd_upscale.py' (str) len=13
extensions\adetailer\scripts\!adetailer.py:939 make_axis_on_xyz_grid
    script.script_class.__module__: 'xyz_grid.py' (str) len=11
2024-04-22 19:49:07,178 - ControlNet - INFO - ControlNet UI callback registered.
Creating model from config: D:\stable-diffusion-webui\configs\v1-inference.yaml
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Startup time: 75.3s (prepare environment: 2.9s, import torch: 4.8s, import gradio: 1.6s, setup paths: 0.7s, initialize shared: 0.3s, other imports: 0.7s, load scripts: 62.6s, scripts before_ui_callback: 0.2s, create ui: 0.7s, gradio launch: 0.2s, add APIs: 0.6s).
Loading VAE weights specified in settings: D:\stable-diffusion-webui\models\VAE\kl-f8-anime2.safetensors
Applying attention optimization: xformers... done.
Model loaded in 4.7s (load weights from disk: 0.4s, create model: 1.3s, apply weights to model: 2.3s, load VAE: 0.2s, calculate empty prompt: 0.2s).
```

It seems to be captured in 1.8.0 style, at least in my environment.

**Comment by w-e-w** [comment]
this dose not seem right
```
Version: v1.9.0-112-gd982c75d
Commit hash: d982c75da9a1647770b3226edaa0a603d20e3a8b
```

1.9 -> 1.9.2 I believe only have about a total of 42 commits
`112` I meant that you have 112 commits on tip of 1.9.0
I also can't find the hush
d982c75da9a1647770b3226edaa0a603d20e3a8b

**Comment by Bing-su** [comment]
Because it's my personal fork. The source code is the same as 1.9.0. 

gitkraken_Ucye27ClNq

**Comment by w-e-w** [comment]
tldr:
it should be fine to NOT merge this
but merging it doesn't hurt and might be slightly better

---

long version

I think I figured out
turns out in this particular issue will only affect a proportion of extensions that checks for `script_class.__module__`
it depends on the timing when the check is performed

the breaking change in one 1.9.0 was is not total as only we have thought
and it doesn't seem to impact this extension in particular

so thers 2 commit that cases this issue
one in 1.9.0
https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/23c06a51ccc54d76db465f8f66a5dcb0f7ca33e5
and between 1.9.0 ~ 1.9.1
https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/15532

the commit in 1.9.0 causes some checks of `if script.script_class.__module__ == "xyz_grid.py":` to break 
and the later commit in 1.9.0~1.9.1 https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/15532
casues every use of `if script.script_class.__module__ == "xyz_grid.py":` to break

we have reverted this in 1.9.2
so the issue should only impact a small portion of people that uses 1.9.0 + dev branch (for this particularly extension)
about 40~ commits 9 days

---

when we were testing I was actually testing with 1.9.0~1.9.1 (dev at the time) I forgot to switch back to 1.9.0 master
I didn't realize the second commitment was the one that causes "every" extension is that checks for `script.script_class.__module__` to break

so when I tested it adetailer xyz on the dev branch with the second commit it was definitely broken
see webui 1.9.0+ e6a8d0b4e65b365a684bd7539cf5b2cd16f9c44c
image
> you can see the if None retrun break point was hit
```py
    if xyz_grid is None:
        return
```

---

so you can decide if you want to merge it or not it doesn't matter that much now
